plugins/woocommerce/src/Internal/StockNotifications/Admin/AdminManager.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\StockNotifications\Admin;
+
+use Automattic\WooCommerce\Internal\StockNotifications\Admin\MenusController;
+use Automattic\WooCommerce\Internal\StockNotifications\Admin\SettingsController;
+use Automattic\Jetpack\Constants;
+
+/**
+ * Admin controller for Customer Stock Notifications.
+ */
+class AdminManager {
+
+	/**
+	 * Initialize admin components.
+	 *
+	 * @internal
+	 *
+	 * @return void
+	 */
+	final public function __construct() {
+
+		// Enqueue scripts.
+		add_action( 'admin_enqueue_scripts', array( $this, 'admin_resources' ), 11 );
+
+		$container = wc_get_container();
+		$container->get( MenusController::class );
+		$container->get( SettingsController::class );
+	}
+
+	/**
+	 * Admin scripts.
+	 *
+	 * @return void
+	 */
+	public static function admin_resources() {
+
+		$screen    = get_current_screen();
+		$screen_id = $screen ? $screen->id : '';
+		$suffix    = Constants::is_true( 'SCRIPT_DEBUG' ) ? '' : '.min';
+		$version   = Constants::get_constant( 'WC_VERSION' );
+
+		wp_register_script( 'wc-admin-customer-stock-notifications', WC()->plugin_url() . '/assets/js/admin/wc-customer-stock-notifications' . $suffix . '.js', array( 'jquery' ), $version, true );
+
+		$params = array(
+			'i18n_wc_delete_notification_warning'       => __( 'Delete this notification permanently?', 'woocommerce' ),
+			'i18n_wc_bulk_delete_notifications_warning' => __( 'Delete the selected notifications permanently?', 'woocommerce' ),
+		);
+
+		/*
+		 * Enqueue specific styles & scripts.
+		 */
+		if ( 'woocommerce_page_wc-customer-stock-notifications' !== $screen_id ) {
+			return;
+		}
+
+		wp_enqueue_script( 'wc-admin-customer-stock-notifications' );
```

Comment:
This is loaded across the admin interface. Let's restrict it to the list table screen only.

Reply:
@xristos3490 This is already enqueued only in the list table screen only: https://github.com/woocommerce/woocommerce/pull/58356/files/245a041fd771e923815576263c23a2839ce3dc9c#diff-38cab0041e3b51210695897b522fa450f6a248c54cc2f4a91e2bb3c1f966f504R55-R56

But will look into using `WCAdminAssets` methods to enqueue this script.

Reply:
Looking at `WCAdminAssets`, there doesn't seem to be a way to use this class for scripts that don't have an `asset.php` file, which I don't think is needed here.

See: https://github.com/woocommerce/woocommerce/blob/4e08bee588c97aec594f0043b9f2c8e94e85a459/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php#L129-L145

Reply:
>This is already enqueued only in the list table screen only: https://github.com/woocommerce/woocommerce/pull/58356/files/245a041fd771e923815576263c23a2839ce3dc9c#diff-38cab0041e3b51210695897b522fa450f6a248c54cc2f4a91e2bb3c1f966f504R55-R56

I missed that, thanks! This looks OK!